packages/astro/src/@types/astro.ts

```diff
@@ -1922,6 +1935,46 @@ export interface AstroUserConfig {
 				origin?: boolean;
 			};
 		};
+
+		/**
+		 * @docs
+		 * @name experimental.rewriting
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.8.0
+		 * @description
+		 *
+		 * Enables the use of rewriting features in Astro pages, Endpoints and Astro middleware:
+		 *
+		 * ```astro
+		 * ---
+		 * // src/pages/dashboard.astro
+		 * if (!Astro.props.allowed) {
+		 * 	return Astro.rewrite("/")
+		 * }
+		 * ---
+		 * ```
+		 *
```

Comment:
Similarly, a statement to say that in your endpoints, use `ctx.rewrite` and in middleware, you can just "return next()" with a route?